stream.go

```diff
@@ -102,23 +103,24 @@ func (stream *Stream) connect() (r io.ReadCloser, err error) {
 	// All but the initial connection will need to regenerate the body
 	if stream.connections > 0 && req.GetBody != nil {
 		if req.Body, err = req.GetBody(); err != nil {
-			return
+			return nil, err
 		}
 	}
 
 	if resp, err = stream.c.Do(&req); err != nil {
-		return
+		return nil, err
 	}
 	stream.connections++
 	if resp.StatusCode != 200 {
 		message, _ := ioutil.ReadAll(resp.Body)
+		_ = resp.Body.Close()
```

Comment:
We were returning the response Body and leaving the user to close it, which didn't seem right given that we had consumed the body.  Now we will just return nil and an error so the user doesn't have to do anything.